### ISSUE TITLE
[ENH] Implementation of MLFlow in sktime

### ISSUE THREAD

**Issue by TNTran92** [alan-turing-institute/sktime#2916]
**Is your feature request related to a problem? Please describe.**
MLFlow is a powerful package itotrack, log, save and deploy ML models. It allows users to save a record of every parameters and  experiments for analysis in a user-friendly manner. Such model can be deploy in the exact state when it was created later on. It would be good to investigate MLFlow's implementation of sktime in the context of benchmarking. 

**Describe the solution you'd like**
A simple implementationof MLFlow to demonstrate that it can perform benchmark using sktime's metrics, dataset and estimators. 

**Describe alternatives you've considered**
OpenML, gluonts, kotsu. Refer to Issue #2777 

**Additional context**

**Comment by TNTran92** [comment]
See sklearn'simplementation examples ([this](https://github.com/mlflow/mlflow/blob/master/examples/sklearn_elasticnet_wine/train.py) and [this](https://github.com/mlflow/mlflow/blob/master/examples/sklearn_elasticnet_diabetes/linux/train_diabetes.py)). If sktime is implemented, it would look very similar. 

In those, they use mlflow to log results and pickle models for later use. The results are then viewable on a web browser (through command `mlflow ui`). All runs on the same codes are saved so you have a record of what you’ve done. From what I've seen, the implementation aka CV, grid search, evaluation, (stuff that `evaluate` does) are done almost entirely in sklearn. So, in short, mlflow - the sklearn implementation looks more like a logging tool than a benchmarking tool.

Benchmarking criteria: Quick and easy way to test multiple estimators on multiple datasets. The “way” here is a framework/procedure that benchmarking is performed on: a standard procedure for how data is passed to estimators, how estimator is evaluated, how CV splitter validates each combination of model and data etc.

In this sense, [kotsu](https://github.com/alan-turing-institute/sktime/issues/2804) is a better fit to perform these tasks. The DataFrame results can then be logged to mlflow and the estimators pickled. 

In conclusion, MlFlow is not an alternative to other benchmarking tool (kotsu) because they are doing different things.

**Comment by aiwalter** [comment]
related #1215
